integration-tests/src/test/java/org/hps/test/it/ReconTest.java

```diff
@@ -0,0 +1,379 @@
+        }
+
+        System.out.println("Reading in test AIDA file: " + testFile.getPath());
+        ITree testTree;
+        try {
+            testTree = af.createTreeFactory().create(testFile.getPath());
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to open test plots");
+        }
+
+        if (printComparisonTable) {
+            printComparisonTable(refTree, testTree);
+        }
+
+        System.out.println("Comparing histograms for: " + this.testClass.getSimpleName());
+        compareTrees(refTree, testTree);
+
+        System.out.println("Done comparing histograms!");
+    }
+
+    /**
+     * Compare reference and test AIDA trees
+     * @param refTree The reference AIDA tree
+     * @param testTree The test AIDA tree
+     */
+    protected void compareTrees(ITree refTree, ITree testTree) {
+        String[] refNames = refTree.listObjectNames("/", true);
+        String[] refTypes = refTree.listObjectTypes("/", true);
+        //System.out.println("AIDA ref file object count: " + refNames.length);
+        for (int i = 0; i < refNames.length; ++i) {
+            String histName = refNames[i];
+            String type = refTypes[i];
+            if (type.startsWith("IHistogram")) {
+                System.out.print(histName + " ... ");
+                if (type.equals("IHistogram1D")) {
+                    compareHistograms((IHistogram1D) refTree.find(histName),
+                            (IHistogram1D) testTree.find(histName));
+                } else if (type.equals("IHistogram2D")) {
+                    compareHistograms((IHistogram2D) refTree.find(histName),
+                            (IHistogram2D) testTree.find(histName));
+                }
+                System.out.println("ok");
+            }
+        }
+    }
+
+    /**
+     * Compare the number of entries, mean and RMS of two 1D histograms
+     * and assert they are equal within the configured tolerance
+     * @param refHist The reference AIDA tree
+     * @param testHist The test AIDA tree
+     */
+    protected void compareHistograms(IHistogram1D refHist, IHistogram1D testHist) {
+        assertEquals("Number of entries is different for: " + refHist.title(), refHist.entries(), testHist.entries());
+        assertEquals("Mean is different for: " + refHist.title(), refHist.mean(), testHist.mean(), tolerance * abs(refHist.mean()));
+        assertEquals("RMS is different for: " + refHist.title(), refHist.rms(), testHist.rms(), tolerance * abs(refHist.rms()));
+    }
+
+    /**
+     * Compare the number of entries, mean and RMS of two 2D histograms
+     * and assert they are equal within the configured tolerance
+     * @param refHist The reference AIDA tree
+     * @param testHist The test AIDA tree
+     */
+    protected void compareHistograms(IHistogram2D refHist, IHistogram2D testHist) {
+        assertEquals("Number of entries is different for: " + refHist.title(), refHist.entries(), testHist.entries());
```

Comment:
Just want to point out that we do have assertions already that have ZERO tolerance which should be able to catch any serious change to the numbers given the ad hoc selection used for these tests.